stl/inc/regex

```diff
@@ -4113,16 +4106,22 @@ void _Parser<_FwdIt, _Elem, _RxTraits>::_ClassRanges() { // check for valid clas
                 _Error(regex_constants::error_range); // set precedes or follows dash
             }
 
-            if (_Flags & regex_constants::collate) { // translate ends of range
-                _Val  = _Traits.translate(static_cast<_Elem>(_Val));
+            _Elem _Chr2 = static_cast<_Elem>(_Val);
+
+            // translate ends of range
+            if (_Flags & regex_constants::icase) {
+                _Chr1 = _Traits.translate_nocase(_Chr1);
+                _Chr2 = _Traits.translate_nocase(_Chr2);
+            } else if (_Flags & regex_constants::collate) {
                 _Chr1 = _Traits.translate(_Chr1);
+                _Chr2 = _Traits.translate(_Chr2);
             }
 
-            if (static_cast<typename _RxTraits::_Uelem>(_Val) < static_cast<typename _RxTraits::_Uelem>(_Chr1)) {
+            if (static_cast<typename _RxTraits::_Uelem>(_Chr2) < static_cast<typename _RxTraits::_Uelem>(_Chr1)) {

```

Comment:
Note that there is still a (pre-existing) bug in this error check when the `collate` flag is set: The bounds should be transformed by `_Traits.transform()` before performing the comparison.

However, the matcher fails to do this `transform()` dance as well, so I think this should rather be fixed for both matcher and parser in a coordinated fashion in a follow-up PR. (But I will adjust the PR if you rather prefer to fix this check completely now.)